### PR TITLE
feat: add rate limiter for L1 deprecation

### PR DESCRIPTION
**Pull request by Theodus**
This adds a per API key rate limiter for the L1 deprecation strategy. Note that this will only be configured on the L1 gateway which only gets 5-15 queries per second total at the time of writing.

### CODE REVIEW

File: gateway-framework/src/auth.rs [hunk 1]
```diff
@@ -41,15 +43,57 @@ pub struct AuthContext {
...
 impl AuthContext {
     /// Parse an authorization token into its corresponding settings, and check that the query
     /// should be handled.
     pub fn check(&self, token: &str, domain: &str) -> anyhow::Result<AuthSettings> {
-        ensure!(!token.is_empty(), "missing bearer token");
+        ensure!(!token.is_empty(), "missing API key");
+
+        if let Some(rate_limiter) = &self.rate_limiter {
+            if rate_limiter.above_limit(token) {
+                bail!("rate limit exceeded");
```
[the remark this answers is not in the document]
Comment (Theodus) [reply gateway-framework/src/auth.rs]: Good suggestion. Done

**Review by IainM22 — APPROVED** [review]

**Comment by LNSD** [comment]
We already have `https://github.com/edgeandnode/gateway/blob/main/gateway-framework/src/http/middleware/rate_limiter.rs`; why don't we use it instead of modifying and adding another "unrelated" responsibility to the `require_auth` module?

**Comment by Theodus** [comment]
@LNSD I don't see how this is "unrelated". We're imposing a limit on API keys similar to checking authorized subgraphs, payment status, etc. So the behavior is all localized, and will be trivial to remove when we no longer need it.

**Review by LNSD — APPROVED** [review]
Ok. Fain enough 👍✅

**Pull request merged**